File: android/src/main/java/com/incode/onboarding_flutter_wrapper/IncodeSdkInitError.java
package com.incode.onboarding_flutter_wrapper;

public enum IncodeSdkInitError {
    SIMULATOR_DETECTED("simulatorDetected"),
    TEST_MODE_ENABLED("testModeEnabled"),
    ROOT_DETECTED("rootDetected"),
    HOOK_DETECTED("hookDetected"),
    UNKNOWN("unknown");

    private final String value;

    IncodeSdkInitError(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: lib/src/models/incode_sdk_init_error.dart
enum IncodeSdkInitError {
  simulatorDetected,
  testModeEnabled,
  rootDetected,
  hookDetected,
  unknown,
}

extension ParseEnumValue on String {
  IncodeSdkInitError? toIncodeSdkInitError() {
    switch (this) {
      case 'simulatorDetected':
        return IncodeSdkInitError.simulatorDetected;
      case 'testModeEnabled':
        return IncodeSdkInitError.testModeEnabled;
      case 'rootDetected':
        return IncodeSdkInitError.rootDetected;
      case 'hookDetected':
        return IncodeSdkInitError.hookDetected;
      case 'unknown':
        return IncodeSdkInitError.unknown;
      default:
        return null;
    }
  }
}
